uk.co.compucorp.civicrm.tasksassignments/js/test/services/document_test.js

```diff
@@ -0,0 +1,77 @@
+/* globals inject */
+/* eslint-env amd, jasmine */
+
+define([
+  'mocks/document',
+  'mocks/contact',
+  'mocks/data/assignment',
+  'common/angularMocks',
+  'tasks-assignments/app'
+], function (documentMock, contactMock, assignmentMock) {
+  'use strict';
+
+  describe('DocumentService', function () {
+    var AssignmentService, ContactService, DocumentService, $q, deferred;
+
+    beforeEach(module('civitasks.appDashboard'));
+    beforeEach(inject(function (_AssignmentService_, _ContactService_, _DocumentService_, _$q_) {
+      AssignmentService = _AssignmentService_;
+      ContactService = _ContactService_;
+      DocumentService = _DocumentService_;
+      $q = _$q_;
+      deferred = $q.defer();
+    }));
+
+    describe('collectAssignmentIds()', function () {
```

Comment:
Unit tests should cover the *public interface* of a service. "collectAssignmentIds()" is a private internal function, it doesn't make sense to make it the subject of the test. This is how the test suite for `DocumentService` should look like
```js
describe('DocumentService', function () {

  describe('CacheContactsAndAssignments', function () {

    it('caches the contacts and assignments of the given documents', function () {

    });

    it('can cache either contacts or assignments exclusively', function () {

    });

    it('returns a promise', function () {

    });
  });
});
```
that's it. Notice that the test is written from the point of view of the "consumer" of the service! It describes the expectation of how the service should behave when used, it doesn't care about the internal functioning